src/engraving/dom/spanner.cpp

```diff
@@ -1511,16 +1511,28 @@ String SpannerSegment::formatBarsAndBeats() const
 
 String SpannerSegment::formatStartBarsAndBeats(const Segment* segment) const
 {
-    std::pair<int, float> barbeat = segment->barbeat();
-    return muse::mtrc("engraving", "Start measure: %1; Start beat: %2")
-           .arg(String::number(barbeat.first), String::number(barbeat.second));
+    EngravingItem::BarBeat barbeat = segment->barbeat();
+    String result = muse::mtrc("engraving", "Start measure: %1").arg(String::number(barbeat.bar));
+
+    if (barbeat.displayedBar != barbeat.bar) {
+        result += u"; " + muse::mtrc("engraving", "Start displayed measure: %1").arg(barbeat.displayedBar);
+    }
+
+    result += u"; " + muse::mtrc("engraving", "Start beat: %1").arg(barbeat.beat);
```

Comment:
I'm not sure how good or whether at all such a combined string works once translated into RTL languages like Hebrew and Arab.

Reply:
I suppose that the correct way to display it for a RTL reader should be C reversed + B reversed + A reversed. That way he would read the measure first, then the displayed measure and then the beat. One solution could be to have an additional translation item with A, B and C as arguments. 

Something like:

```C++
    QString endMeasure = muse::qtrc("engraving", "End measure: %1").arg(String::number(endBarbeat.bar));
    QString endDisplayedMeasure = "";
    if (endBarbeat.displayedBar != endBarbeat.bar) {
        endDisplayedMeasure = muse::qtrc("engraving", "; End displayed measure: %1").arg(endBarbeat.displayedBar);
    }
    QString endBeat = muse::qtrc("engraving", "; End beat: %1").arg(endBarbeat.beat);
    QString endResult = muse::qtrc("engraving", "%1 %2 %3").arg(String::number(endMeasure), String::number(endDisplayedMeasure), String::number(endBeat));
```

Then, the translation team can chose what order to use for that string. On the other hand it might be a bit harder to maintain and I am not particularly fond of having a semi-colon separator as part of the string itself.

Reply:
You could do something with `if (qApp->layoutDirection() == Qt::LayoutDirection::RightToLeft)`

Reply:
A lot of similar combined strings exist in the code. After discussion with @cbjeukendrup and @wizofaus in the discord dev channel, I think that it would be better to fix the issue for the whole application in another task instead of patching this one only.

Reply:
Plus that I don't think it is necessary at all: afaik, in RTL languages, the characters are _rendered_ from right to left, but _logically_ they appear "from left to right" in the computer memory, so string concatenation should Just Work.